tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -1138,19 +1139,23 @@ class SerializableModelLibraryGenerator {
     return fields
         .where((field) => field.shouldIncludeField(serverCode))
         .map((field) {
-      var type = field.type.reference(
+      var fieldType = field.type.reference(
         serverCode,
         nullable: true,
         subDirParts: classDefinition.subDirParts,
         config: config,
       );
 
+      var isInheritedField = classDefinition.inheritedFields.contains(field);
+
+      var type = field.type.nullable && isInheritedField
+          ? refer('Object?')
+          : fieldType;
+
```

Comment:
Suggestion: Could we add some test for this change?

Reply:
Yes I'll add one in `server_code_generator > inheritance_class_test.dart` checking if 
inherited-nullable => `Object?` 
local-nullable => typed (int?, String?, ...)